File: 12.整数转罗马数字.cs
/*
 * @lc app=leetcode.cn id=12 lang=csharp
 *
 * [12] 整数转罗马数字
 */

// @lc code=start
public class Solution
{
    public string IntToRoman(int num)
    {
        string res = "";
        string[] romanStrs = new string[] { "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I" };
        int[] romanNums = new int[] { 1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1 };
        for (int i = 0; i < romanStrs.Length;)
        {
            if (num >= romanNums[i])
            {
                res += romanStrs[i];
                num -= romanNums[i];
            }
            else
            {
                i++;
            }
        }
        return res;
    }
}
// @lc code=end

File: 12.整数转罗马数字.rs
/*
 * @lc app=leetcode.cn id=12 lang=rust
 *
 * [12] 整数转罗马数字
 */

// @lc code=start
impl Solution {
    pub fn int_to_roman(num: i32) -> String {
        let mut res = String::new();
        let roman_strs = vec![
            "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I",
        ];
        let roman_nums = vec![1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
        let mut new_num = num;
        let mut i = 0;

        while i < 13 {
            if new_num >= roman_nums[i] {
                res+=roman_strs[i]
                new_num -= roman_nums[i];
            }else{
                i+=1;
            }
        }
        res
    }
}
// @lc code=end
